scipy/stats/kde.py

```diff
@@ -102,17 +106,24 @@ class gaussian_kde(object):
 
     Scott's Rule [1]_, implemented as `scotts_factor`, is::
 
-        n**(-1./(d+4)),
+        neff**(-1./(d+4)),
 
-    with ``n`` the number of data points and ``d`` the number of dimensions.
-    Silverman's Rule [2]_, implemented as `silverman_factor`, is::
+    with ``neff`` the effective number of data points and ``d`` the number of
+    dimensions.  Silverman's Rule [2]_, implemented as `silverman_factor`, is::
 
-        (n * (d + 2) / 4.)**(-1. / (d + 4)).
+        (neff * (d + 2) / 4.)**(-1. / (d + 4)).
 
     Good general descriptions of kernel density estimation can be found in [1]_
     and [2]_, the mathematics for this multi-dimensional implementation can be
     found in [1]_.
 
+    With a set of weighted samples, the effective number of datapoints ``neff``
+    is defined by:
+
+        neff = sum(weights)^2 / sum(weights^2)
```

Comment:
this doesn't look right, should it be `sum(weights)^2 / sum(weights^2) * n`?

Reply:
It's correct:

* Take the default case of equal weighting, e.g: `weights=numpy.ones(n)`
* Here, `neff=n^2/n=n`, i.e. the right effective number
* `sum(weights)^2 / sum(weights^2)` is invariant under rescaling of the weights (`weights`->`k*weights`), so this answer is true for any set of equal weights.
* Your modification would break the default case, giving `neff=n**2`

Reply:
clear, thanks.